services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSpec.groovy

```diff
@@ -0,0 +1,463 @@
+package org.openkilda.functionaltests.spec.switches
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.model.MeterId.MAX_SYSTEM_RULE_METER_ID
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.PathHelper
+import org.openkilda.functionaltests.helpers.SwitchHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.command.switches.DeleteRulesAction
+import org.openkilda.model.Cookie
+import org.openkilda.model.SwitchId
+import org.openkilda.northbound.dto.switches.SwitchValidationResult
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import groovy.transform.Memoized
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+
+@Narrative("""This test suite checks the switch validate feature on a flow that contains 2+ switches.
+Description of fields:
+- missing - those meters/rules, which are NOT present on a switch, but are present in db
+- misconfigured - those meters which have different value (on a switch and in db) for the same parameter
+- excess - those meters/rules, which are present on a switch, but are NOT present in db
+- proper - meters/rules values are the same on a switch and in db
+""")
+class SwitchValidationSpec extends BaseSpecification {
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Switch validation is able to store correct information about a flow in the 'proper' section"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
```

Comment:
I think this test should work with nonNeighboring switches and add switch validation for transit switches. transit switches have special type of rules and no meters, verifying switch validation for them is a must

Reply:
it was ironed out(internally discussed)